wagtail/core/models/__init__.py

```diff
@@ -450,7 +450,12 @@ def is_site_root(self):
 
         This includes translations of site root pages as well.
         """
-        return Site.objects.filter(root_page__translation_key=self.translation_key).exists()
+        if hasattr(self, "_is_site_root"):
```

Comment:
I'm not enormously keen on the Page model code having special knowledge of an optimisation that only exists in wagtail.admin view code. Perhaps we could make `prefetch_workflow_states`, `annotate_approved_schedule` and `annotate_site_root_state` into methods of PageQuerySet instead (they'll probably need the imports of Site / WorkflowState / PageRevision to be inside the function bodies to avoid a circular import, but that should be fine), and add a comment here to clarify that that's where this attribute comes from?

Reply:
Sure, this is how had it [originally](https://github.com/wagtail/wagtail/pull/7318/commits/870949cd4074a09f8c4b86df309e6046d209e4f2) but Karl [suggested otherwise](https://github.com/wagtail/wagtail/pull/7318#discussion_r667759669), as they're likely only to be used in this view.

Reply:
Sorry I missed this bit!

If we move them back, could we make it clear by the naming/docstrings which properties on the ``Page`` model are optimised by each method?

Reply:
I've moved these back to the queryset and clarified where they are set in docstrings/comments in both directions.